### PR TITLE
install cron on fedora

### LINKED ISSUE #201: cron job creation operations fail on Fedora
Fedora no longer has a crontab file:

```
      Recipe: test::default
         * chef_client_updater[Install Chef 13.10.0] action update
           * chef_gem[mixlib-install] action install
             - install version ~> 3.11 of package mixlib-install
           - remove /opt/chef.upgrade from previous Chef Infra Client run
           * chef_gem[mixlib-install] action install (up to date)
           * cron[chef_client_updater] action delete

             ================================================================================
             Error executing action `delete` on resource 'cron[chef_client_updater]'
             ================================================================================

             Chef::Exceptions::Cron
             ----------------------
             Error determining state of chef_client_updater, error: No such file or directory - crontab

             Resource Declaration:
             ---------------------
             # In /tmp/kitchen/cache/cookbooks/chef_client_updater/providers/default.rb

             252:   cron 'chef_client_updater' do
             253:     action :delete
             254:   end unless platform_family?('windows')
             255: end
             256:

             Compiled Resource:
             ------------------
             # Declared in /tmp/kitchen/cache/cookbooks/chef_client_updater/providers/default.rb:252:in `cleanup'

             cron("chef_client_updater") do
        action [:delete]
        default_guard_interpreter :default
        minute "*"
        hour "*"
        day "*"
        month "*"
        weekday "*"
        user "root"
        declared_type :cron
        cookbook_name "test"
             end

             System Info:
             ------------
             chef_version=13.10.0
             platform=fedora
             platform_version=32
             ruby=ruby 2.4.4p296 (2018-03-28 revision 63013) [x86_64-linux]
             program_name=/opt/chef/bin/chef-solo
             executable=/opt/chef/bin/chef-solo


           ================================================================================
           Error executing action `update` on resource 'chef_client_updater[Install Chef 13.10.0]'
           ================================================================================

           Chef::Exceptions::Cron
           ----------------------
           cron[chef_client_updater] (/tmp/kitchen/cache/cookbooks/chef_client_updater/providers/default.rb line 252) had an error: Chef::Exceptions::Cron: Error determining state of chef_client_updater, error: No such file or directory - crontab

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/test/recipes/default.rb

             1: chef_client_updater "Install Chef #{node['chef_client_updater']['version']}" do
             2:   channel 'stable'
             3:   version node['chef_client_updater']['version']
             4:   post_install_action 'exec'
             5:   install_timeout 599
             6: end

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/test/recipes/default.rb:1:in `from_file'

           chef_client_updater("Install Chef 13.10.0") do
             action [:update]
             updated true
             updated_by_last_action true
             default_guard_interpreter :default
             declared_type :chef_client_updater
             cookbook_name "test"
             recipe_name "default"
             channel "stable"
             version "13.10.0"
             post_install_action "exec"
             install_timeout 599
             rubygems_url "https://www.rubygems.org"
           end

           System Info:
           ------------
           chef_version=13.10.0
           platform=fedora
           platform_version=32
           ruby=ruby 2.4.4p296 (2018-03-28 revision 63013) [x86_64-linux]
           program_name=/opt/chef/bin/chef-solo
           executable=/opt/chef/bin/chef-solo


       Running handlers:
       [2020-05-27T20:41:43+00:00] ERROR: Running exception handlers
       [2020-05-27T20:41:43+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2020-05-27T20:41:43+00:00] ERROR: Exception handlers complete
       [2020-05-27T20:41:43+00:00] ERROR: Exception handlers complete
       Chef Client failed. 1 resources updated in 03 seconds
       [2020-05-27T20:41:43+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2020-05-27T20:41:43+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2020-05-27T20:41:43+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2020-05-27T20:41:43+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2020-05-27T20:41:43+00:00] FATAL: Chef::Exceptions::Cron: chef_client_updater[Install Chef 13.10.0] (test::default line 1) had an error: Chef::Exceptions::Cron: cron[chef_client_updater] (/tmp/kitchen/cache/cookbooks/chef_client_updater/providers/default.rb line 252) had an error: Chef::Exceptions::Cron: Error determining state of chef_client_updater, error: No such file or directory - crontab
       [2020-05-27T20:41:43+00:00] FATAL: Chef::Exceptions::Cron: chef_client_updater[Install Chef 13.10.0] (test::default line 1) had an error: Chef::Exceptions::Cron: cron[chef_client_updater] (/tmp/kitchen/cache/cookbooks/chef_client_updater/providers/default.rb line 252) had an error: Chef::Exceptions::Cron: Error determining state of chef_client_updater, error: No such file or directory - crontab
```

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: nirbhay1997 <nirbhay@msystechnologies.com>

### Description

Installation of cron on fedora as by default it is not installed on the newer versions
### Issues Resolved
#201 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: providers/default.rb [hunk 1]
```diff
@@ -298,6 +298,25 @@ def upgrade_start_time
...
+  msg = ['Something went wrong while installing cronie.']
+  unless cronie_install.stdout.empty?
+    msg << 'STDOUT:'
+    msg << cronie_install.stdout
+  end
+  unless cronie_install.stderr.empty?
+    msg << 'STDERR:'
+    msg << cronie_install.stderr
+  end
+  Chef::Log.warn msg.join("\n")
```
Comment (Nirbhay1997) [inline providers/default.rb:316]: This part will log the error messages if some error occurred while installing  cron

File: providers/default.rb [hunk 2]
```diff
@@ -639,6 +658,7 @@ def license_acceptance!
 action :update do
   begin
     load_prerequisites!
+    prepare_fedora! if platform? 'fedora'
```
Comment (Nirbhay1997) [inline providers/default.rb:660]: gated only for fedora

File: providers/default.rb [hunk 3]
```diff
@@ -298,6 +298,25 @@ def upgrade_start_time
   shifted_time.strftime('%H:%M')
 end
 
+def prepare_fedora!
+
+  cronie_install = shell_out("dnf install cronie -y", timeout: 60)
```
Comment (Nirbhay1997) [inline providers/default.rb]: the command in the shellout will install cronie with all the dependencies

File: providers/default.rb [hunk 4]
```diff
@@ -299,21 +299,8 @@ def upgrade_start_time
...
-    msg << 'STDOUT:'
-    msg << cronie_install.stdout
-  end
-  unless cronie_install.stderr.empty?
-    msg << 'STDERR:'
-    msg << cronie_install.stderr
-  end
-  Chef::Log.warn msg.join("\n")
+  shell_out!('dnf install cronie -y', timeout: 60)
+  Chef::Log.debug 'Cron installed Successfully'
```
Comment (Nirbhay1997) [inline providers/default.rb:303]: Installs cron package with all the dependencies and logs out the successful message and  raises an exception if anything goes wrong

**Comment by tas50** [comment]
This feels like the wrong approach on Fedora. Fedora doesn't come with Cron and future versions of RHEL will be the same. These have moved over to systemd timers and we shouldn't be adding cron to hosts where it doesn't belong.

**Pull request closed without merging**